src/main/java/cn/nukkit/level/Level.java

```diff
@@ -2939,7 +2939,12 @@ public Item useItemOn(Vector3 vector, Item item, BlockFace face, UseItemData dat
 
         boolean isPlayerInput = data.triggerType == UseItemData.TriggerType.PLAYER_INPUT;
 
-        if (player != null && isPlayerInput) {
+        // TriggerType == SIMULATION_TICK when a player spam-uses bone meal on a fertilizable block
+        // therefore isPlayerInput check has to be overriden when bone meal is used
+        boolean allowFertilizerOverride = item.isFertilizer();
+        boolean isValidTrigger = !isPlayerInput || allowFertilizerOverride;
```

Comment:
Should be `isPlayerInput || allowFertilizerOverride`, maybe simplify to `isPlayerInput || item.isFertilizer()`